### PR TITLE
Roles

### PR DESCRIPTION
**Pull request by IrinaLysenko21**

### CODE REVIEW

**Comment by neprotod** [comment]
test

**Comment by IrinaLysenko21** [comment]
pull request

File: src/components/user/model.js [hunk 1]
```diff
@@ -85,6 +85,17 @@ module.exports = {
...
     return result;
   },
+  /**
+   * Find user by email
+   *
+   * @param {String} email user email
+   * @param {String} password user password
+   * @return {{}} found user in db
+   */
+  async getUserByEmailAndPassword(email, password) {
```
[the remark this answers is not in the document]
Comment (IrinaLysenko21) [reply src/components/user/model.js]: Функцию удалила

File: src/components/user/model.js [hunk 2]
```diff
@@ -101,38 +106,90 @@ module.exports = {
...
    *
-   * @param {String} id
-   * @return {{}} found a role
+   * @param {String} id role id
+   * @return {{}} deleted role from db
    */
-  async getRole(id) {
-    const result = await Role.findById(id);
+  async deleteRole(id) {
+    const result = await Role.findByIdAndDelete(id);
```
[the remark this answers is not in the document]
Comment (IrinaLysenko21) [reply src/components/user/model.js]: deleteRole controller deletes role not just from db, but from user object too

async deleteRole(req, res) {
    try {
      const userId = req.session.id_user;
      const role = await Role.deleteRole(req.params.id);
      await User.deleteRoleFromUser(userId, role.id);

      res.status(200).json(role);
    } catch (e) {
      return res.status(400).json({errors: e});
    }
  },

File: src/components/user/model.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (IrinaLysenko21) [reply src/components/user/model.js]: function is deleted

**Pull request merged**